lang/rust/rust.talon

```diff
@@ -35,21 +46,63 @@ state enum <user.text>:
 
 toggle use: user.code_toggle_libraries()
 
-state pub: "pub "
-state pub crate: "pub(crate) "
-state dyn: "dyn "
-state imp: "impl "
+## Simple aliases
+borrow: "&"
+borrow mutable: "&mut "
+state (pub | public): "pub "
+state (pub | public) crate: "pub(crate) "
+state (dyn | dynamic): "dyn "
+state constant: "const "
+state (funk | func | function): "fn "
+state (imp | implements): "impl "
 state let mute: "let mut "
 state let: "let "
-state mute: "mut "
-state ref mute: "ref mut "
+state (mute | mutable): "mut "
+state ref (mute | mutable): "ref mut "
 state ref: "ref "
 state trait: "trait "
 state match: "match "
-op arrow: " -> "
-op dub arrow: " => "
-state use: "use "
+state static: "static "
+self taught: "self."
+state use: user.code_import()
 
 use <user.code_libraries>:
     user.code_insert_library(code_libraries, "")
     key(semicolon enter)
+
+## specialist flow control
+if let some: user.code_insert_if_let_some()
+if let error: user.code_insert_if_let_error()
+
+## rust centric synonyms
+is (some|sum): user.code_insert_is_not_null()
```

Comment:
Is `sum` here for Dragon? If not and Dragon works without it, it'd be clearer to just have `is some`.

Reply:
done